cpp/src/umap/umap.cu

```diff
@@ -15,117 +15,115 @@
  */
 
 #include <cuml/manifold/umapparams.h>
+#include <cuml/manifold/common.hpp>
 #include <cuml/manifold/umap.hpp>
 #include "runner.cuh"
 
 #include <iostream>
 
 namespace ML {
+namespace UMAP {
 
-static const int TPB_X = 256;
+static const int TPB_X = 32;
```

Comment:
This sets the threads per block to the size of a warp. This was set to 32 originally and I had changed it to 256 at some point. In general, when a kernel is just maximizing parallelism by launching a bunch of independent threads, I've been setting this to 32 (less threads per block, thus less memory needed to be shared across the block) so that more blocks execute independently. Otherwise, I'm not sure this matters much and I didn't find it to affect the runtime when I was profiling for the paper.

Reply:
maybe then it is better to use `WarpSize` from [here](https://github.com/rapidsai/raft/blob/branch-0.17/cpp/include/raft/cuda_utils.cuh#L100), instead of specifying explicit value?

Reply:
@teju85 I'm hoping to get your thoughts about my reasoning above. I was thinking that thread-level parallelism is going to be very low with independent threads that have 1) many reads/writes to global memory and 2) many branching conditionals, thus scheduling more blocks is going to have more parallelism. After talking w/ a few folks and doing a little bit of research, I believe the parallelism stays the same, but now less blocks can be scheduled on each SM.

Wondering now if the reason I don't see a huge difference in UMAP is that the KNN is such a huge bottleneck that I have only profiled with moderate data sizes (500k or less).